src/mappings/utils/vaultThresholds.ts

```diff
@@ -0,0 +1,155 @@
+import { SubstrateBlock } from "@subsquid/substrate-processor";
+import { Ctx } from "../../processor";
+import { 
+    VaultRegistryLiquidationCollateralThresholdStorage,
+    VaultRegistryPremiumRedeemThresholdStorage,
+    VaultRegistrySecureCollateralThresholdStorage,
+ } from "../../types/storage";
+import { CurrencyId as CurrencyIdv1 } from "../../types/v1";
+import { VaultCurrencyPair as VaultCurrencyPairv3 } from "../../types/v3";
+import { VaultCurrencyPair as VaultCurrencyPairv6 } from "../../types/v6";
+import { 
+    Key_ExchangeRate as Key_ExchangeRatev15,
+    VaultCurrencyPair as VaultCurrencyPairv15,
+    CurrencyId as CurrencyIdv15,
+} from "../../types/v15";
+import {
+    Key_ExchangeRate as Key_ExchangeRatev17,
+    VaultCurrencyPair as VaultCurrencyPairv17,
+    CurrencyId as CurrencyIdv17,
+} from "../../types/v17";
+import { VaultCurrencyPair as VaultCurrencyPairv1020000 } from "../../types/v1020000";
+import { VaultCurrencyPair as VaultCurrencyPairv1021000 } from "../../types/v1021000";
+import { CollateralThreshold, Currency } from "../../model";
+import { BigDecimal } from "@subsquid/big-decimal";
+
+
+export async function getLiquidationThreshold(
+    ctx: Ctx,
+    block: SubstrateBlock,
+    key: CurrencyIdv1 | VaultCurrencyPairv3 | VaultCurrencyPairv6   | VaultCurrencyPairv15 
+    | VaultCurrencyPairv17 | VaultCurrencyPairv1020000 | VaultCurrencyPairv1021000
+)
+{
+    const rawLiquidationThreshold = new VaultRegistryLiquidationCollateralThresholdStorage(ctx, block);
+    let value;
+    if (rawLiquidationThreshold.isV3)
```

Comment:
Please do use curly braces for the blocks, even if it's only one line as it improves readability.